src/kernel/caches.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_CACHES_H
+#define BITCOIN_KERNEL_CACHES_H
+
+#include <algorithm>
+#include <cstdint>
+
+//! Suggested default amount of cache reserved for the kernel (MiB)
+static constexpr int64_t DEFAULT_KERNEL_CACHE{450};
+//! Max memory allocated to block tree DB specific cache (MiB)
+static constexpr int64_t MAX_BLOCK_DB_CACHE{2};
+//! Max memory allocated to coin DB specific cache (MiB)
+static constexpr int64_t MAX_COINS_DB_CACHE{8};
+
+namespace kernel {
+struct CacheSizes {
+    size_t block_tree_db;
+    size_t coins_db;
+    size_t coins;
```

Comment:
I think I'm ~0 on this change. I follow @ryanofsky's [reasoning](https://github.com/bitcoin/bitcoin/pull/31483#pullrequestreview-2509692136), but it also goes against the work done e.g. in https://github.com/bitcoin/bitcoin/pull/23962. If `kernel::CacheSizes` becomes part of the public kernel interface, then keeping these members an `int64_t` might be more prudent, as per e.g. the link in [this comment](https://github.com/bitcoin/bitcoin/pull/23962#issuecomment-1005494716)? But of course, `size_t` is a pretty natural fit for a size type.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/31483#discussion_r1890151750

I agree with changes in the issues you linked to, but I don't think the reasoning in those cases carries over to this one. int32_t is a good choice for transaction sizes and weights, because we know the sizes of transactions are bounded and can't get very big, so it makes sense to use a small type, and we want the bounds to be platform independent so it makes to use fixed width types, and because they are numbers we do arithmetic with, and signed types tend to be safer than unsigned types for arithmetic.

I think size_t is a better choice for the cache size variables in this PR, because these values are eventually used to allocate memory so making them size_t from the beginning is clearer and avoids the need for casts and conversions later on. Also size_t is the standard type to use for this purpose so it's more idiomatic c++. Also maximum cache sizes do depend on the platform, so size_t a safe choice because it is guaranteed to be able to represent the maximum size of any allocated object on a platform. I do think a downside of choosing size_t is that it is unsigned and unsigned types have less intuitive overflow behavior, but if the values are just going to get casted to size_t later anyway, the overflow behavior is there no matter what.

Reply:
Yeah, thinking about it more I agree with all of your points. I was a bit alarmed by all the comparison operations we were doing with these (now) `size_t` types, but they almost all happen inside the constructor. Once the sizes are determined, they don't seem likely to be used for arithmetic operations or comparisons, just for allocating memory.

I think the benefits you outline outweigh the potential downsides, so I'm happy with the new approach.